code/game/objects/structures/gym.dm

```diff
@@ -0,0 +1,216 @@
+
+/obj/structure/weightmachine/weightlifter/animate_machine(mob/living/user)
+	var/mutable_appearance/swole_overlay = mutable_appearance(icon, "fitnessweight-w", ABOVE_MOB_LAYER)
+	overlays += swole_overlay
+	user.pixel_z = 5
+	for(var/reps in 1 to 6)
+		if(user.loc != src.loc)
+			break
+		for(var/innerReps = max(reps, 1), innerReps > 0, innerReps--)
+			sleep(3 DECISECONDS)
+			animate(user, pixel_z = (user.pixel_z == 3) ? 5 : 3, time = 3)
+		playsound(user, 'sound/effects/spring.ogg', 40, TRUE, 2)
+	animate(user, pixel_z = 2, time = 3, delay = 3)
+	sleep(6 DECISECONDS)
+	overlays -= swole_overlay
+
+// Treadmill
+
+/obj/structure/machinery/treadmill
+	icon = 'icons/obj/structures/treadmill.dmi'
+	icon_state = "back_off"
+	name = "treadmill"
+	desc = "A treadmill used for cardio related exercise."
+	layer = CONVEYOR_LAYER // so they appear under stuff
+	anchored = TRUE
+	idle_power_usage = 10
+	active_power_usage = 50
+	var/datum/weakref/second_half_ref
+
+/obj/structure/machinery/treadmill/console
+	icon_state = "console_nopower"
+
+/obj/structure/machinery/treadmill/Destroy()
+	stop_processing()
+	if(second_half_ref)
+		var/obj/structure/machinery/treadmill/second_half = second_half_ref?.resolve()
+		second_half.second_half_ref = null
+		qdel(second_half)
+		second_half_ref = null
+	return ..()
+
+/obj/structure/machinery/treadmill/update_icon()
+	. = ..()
+	if(machine_processing && !(stat & NOPOWER))
+		icon_state = "back_on"
+	else
+		icon_state = "back_off"
+
+/obj/structure/machinery/treadmill/Crossed(O)
+	. = ..()
+	var/mob/entering_mob = O
+	if(!istype(entering_mob))
+		return
+	entering_mob.pixel_z += pixel_y + 2
+
+/obj/structure/machinery/treadmill/Uncrossed(O)
+	. = ..()
+	var/mob/exiting_mob = O
+	if(!istype(exiting_mob))
+		return
+	exiting_mob.pixel_z -= pixel_y + 2
+
+/obj/structure/machinery/treadmill/proc/toggle_running(force_state = null)
+	var/to_process = FALSE
+	if(!isnull(force_state))
+		to_process = force_state
+	else
+		to_process = !machine_processing
+
+	if(to_process)
+		start_processing()
+	else
+		stop_processing()
+
+	update_icon()
+	update_use_power(machine_processing ? USE_POWER_ACTIVE : USE_POWER_IDLE)
+
+/obj/structure/machinery/treadmill/process()
+	if(inoperable())
+		return PROCESS_KILL
+	var/items_moved = 0
+	for(var/atom/movable/A in loc.contents - src)
+		if(!A.anchored)
+			if(A.loc == src.loc)
+				INVOKE_NEXT_TICK(src, PROC_REF(move_thing), A) // to prevent instant movement
+				items_moved++
+		if(items_moved >= 10)
+			break
+	return 1
+
+/obj/structure/machinery/treadmill/proc/move_thing(atom/thing)
+	step(thing, turn(dir, 180))
+
+/obj/structure/machinery/treadmill/console/Initialize(mapload, ...)
+	. = ..()
+	var/turf/T = get_step(src, turn(dir, 180))
+	for(var/obj/O in T)
+		if(O.type == /obj/structure/machinery/treadmill)
+			var/obj/structure/machinery/treadmill/sec_half = O
+			second_half_ref = WEAKREF(sec_half)
+			sec_half.second_half_ref = WEAKREF(src)
+			return
```

Comment:
need to replace the single letter variables https://github.com/cmss13-devs/cmss13/blob/master/.github/guides/STYLES.md#dont-use-abbreviations